### ISSUE TITLE
Add "getting started" usage examples 

### ISSUE THREAD

**Issue by brandonaut** [tianocore/edk2-pytool-extensions#87]
Looking around the features, I'm really excited to give these tools a try in my own EDK2 projects, but I cannot figure out what steps I need to perform in order to integrate them into my codebase. 

[docs/using.md](https://github.com/tianocore/edk2-pytool-extensions/blob/master/docs/using.md) tantalizingly states:

> You are one settings file away from a great platform building experience.

But I cannot figure out what settings file it is referring to, how to create it, or where to put it.

Basically, it would be really helpful to have a getting started guide that walks through how to take an existing EDK2 project and adapt it to use these tools.

**Comment by matthewfcarlson** [comment]
Hey @brandonaut, thanks for checking out edk2-pytools (stuart). I agreed- our documentation on how to use stuart in an existing code tree are a little lacking. We currently have a fork of the EDK2 tree that is setup to CI (https://github.com/spbrogan/edk2-staging). CI and Platform builds are pretty different, but the mechanisms are similar enough that it should give you a general feel for what it should do.

That said, I completely agree. I've recently been moving our internal platforms to use edk2-pytools and a document that details the process would be helpful. I will start on that. Do you have any suggestions or feedback on the docs so far? Would a sample platform like NXP or Raspberry Pi be helpful or would a more general tutorial of how we converted our own platforms be more helpful?

**Comment by brandonaut** [comment]
Thanks for pointing me to the fork, I'll start looking through it.

A general tutorial of how you converted your own platforms would be very helpful. I'm still struggling to understand what invocables and plugins are, and when I need to care about them.

At first I thought that a settings file would just be a set of switches to configure for a project (in JSON or somesuch), but it seems to be a full set of classes that need to be implemented for every action you want stuart to be able to perform on the codebase.

**Comment by matthewfcarlson** [comment]
Let me know if you have any questions!

I'll work on that! I don't have any fully open source versions of our internal platforms, but I can move one the systems in edk2-platforms that I'm familiar with to edk2 stuart.

Yeah- the settings is a lightweight python file. This allows maximum flexibility for platforms and codebases to have conditional settings and built it into pipelines and build systems. You could likely create a settings file that would load settings from JSON and use it on all your projects.
